file_transfer_api/src/file_manager/aws_file_manager.py

```diff
@@ -0,0 +1,110 @@
+from pathlib import Path
+from uuid import uuid4
+
+from src.file_manager.abstract_file_manager import AbstractFileManager
+import boto3
+import os
+from botocore.exceptions import NoCredentialsError
+from src.exceptions.file_exceptions import (FileDownloadError, FileUploadError, FileDeleteError, FileDoesNotExistError,
+                                            FileUpdateError)
+from src.file_manager.utils.url_utils import URLPath
+from typing import IO, Optional
+
+# Get download location from the environment
+download_path = os.getenv("DOWNLOAD_DIRECTORY", default="data/downloads")
+
+
+class AWSFileManager(AbstractFileManager):
+    def __init__(self, target_bucket: str, s3_client: Optional[boto3.client] = None):
+        """Initialises the AWS file manager.
+
+        Args:
+            target_bucket: Name of the target bucket in AWS S3.
+            s3_client: AWS S3 client.
+        """
+        if s3_client is None:
+            self.client = boto3.client("s3")
+        else:
+            self.client = s3_client
```

Comment:
You could simplify this to:
```suggestion
    def __init__(self, target_bucket: str, s3_client: Optional[boto3.client] = boto3.client("s3")):
        """Initialises the AWS file manager.

        Args:
            target_bucket: Name of the target bucket in AWS S3.
            s3_client: AWS S3 client.
        """
        self.client = s3_client
```